src/histogram/Histogram.cpp

```diff
@@ -233,37 +235,51 @@ Histogram::GetSlice(const std::map<std::string,size_t>& fixedBins_) const{
 			      << it->first << "!");
     }
 
-    // want a 1d slice, so check the dimensions
-    if (fixedBins_.size() != fNDims-1)
-        throw DimensionError("Histogram::GetSlice", fNDims-1, fixedBins_.size());
-  
-    // work out which axis is the slice in (i.e. the not fixed one)
-    std::string newAxisName;
-    size_t newAxisIndex;
-    std::vector<size_t> sliceIndices(fNDims);
+    // want to support multidimensional slices 
+    if (fixedBins_.size() >= fNDims)
+        throw DimensionError("Histogram::GetSlice", fNDims -1, fixedBins_.size());
+
+    // create vectors to hold the slicing idxs 
+    std::vector<std::string> newAxisNames; 
+    std::vector<size_t> newAxisIndexes;
+    std::vector<size_t> sliceIndices(fNDims); 
   
     for(std::vector<std::string>::const_iterator it = allAxisNames.begin(); it!=allAxisNames.end(); it++){
+        
+        // goes through entire fixedBins_ vec and reaches end without finding axis in fixedBins 
         if(fixedBins_.find(*it) == fixedBins_.end()){
 	    //this is the axis we want the slice in
-	    newAxisName = *it;
-	    newAxisIndex = fAxes.GetAxisIndex(*it);
-	}else{
+        newAxisNames.push_back(*it); 
+        newAxisIndexes.push_back(fAxes.GetAxisIndex(*it));
+	    } else{
 	    //record the indices for the fixed bins in other dimensions
 	    sliceIndices[fAxes.GetAxisIndex(*it)] = fixedBins_.at(*it);
-	}
+	    }
     }
  
-    // new histogram
-    AxisCollection newAxes;
-    newAxes.AddAxis(fAxes.GetAxis(newAxisIndex));
-    Histogram slice(newAxes);
+    // new histogram loop 
+    AxisCollection newAxes; 
+    std::vector<size_t> binAxisBins;  
+    for(int i = 0; i < newAxisNames.size(); i++){
```

Comment:
You're doing a comparison between a signed integer `i` and an unsigned one `newAxisNames.size()`, which is something to avoid (it brings up a compiler warning!)
Have `i` be the `size_t` type instead.